docs/api/features/placement.md

```diff
@@ -1,3 +1,137 @@
+filters"](https://docs.docker.com/swarm/scheduler/filter/#use-an-affinity-filter)
+where node selection is described in terms of existing containers. For example:
+
+    docker run --name db1 -e 'affinity:container!=db0' -d mysql
+    #                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    # Run a mysql instance that is NOT on the same node as container 'db0'.
+
+For the same reasons, node selection can matter on Triton. Triton's Docker
+implements the same affinity filters as documented for Docker Swarm, with the
+difference that you don't need to setup a Swarm cluster.
+
+
+### Affinity filter syntax
+
+An affinity filter is an argument to `docker run` or `docker create` using
+either (a) environment variables:
+
+    docker run -e 'affinity:<filter>' ...
+
+or (b) labels. (Note: At the time of writing the label syntax was in Docker
+Swarm [code](https://github.com/docker/swarm/blob/d9beef7/cluster/config.go#L83-L86),
+but not [documented](https://docs.docker.com/swarm/scheduler/filter/).)
+
+    docker run --label 'com.docker.swarm.affinities=["<filter>",...]' ...
+
+
+A `<filter>` is one of the following. (Note: Swarm defines "image filters"
+which don't apply for Triton because any image in the datacenter is available
+for any node.)
+
+- A container filter: `container<op><value>`
+- A label filter: `<label><op><value>`
+
+`<op>` is one of:
+
+- `==`: The new container must be on the same node as the container(s)
+  identified by `<value>`.
+- `!=`: The new container must be on a different node as the container(s)
+  identified by `<value>`.
+- `==~`: The new container should be on the same node as the container(s)
+  identified by `<value>`. I.e. this is a best effort or "soft" rule.
+- `!=~`: The new container should be on a different node as the container(s)
+  identified by `<value>`. I.e. this is a best effort or "soft" rule.
+
+`<value>` is an exact string, simple `*`-glob, or regular expression to
+match against container names or IDs, or against the given label name.
+(See also the [Docker Swarm filter expression
+documentation](https://docs.docker.com/swarm/scheduler/filter/#how-to-write-filter-expressions)).
+
+Some examples:
+
+    # Run on the same node as silent_bob:
+    docker run -e 'affinity:container==silent_bob' ...
+
+    # Same, using the label syntax:
+    docker run --label 'com.docker.swarm.affinities=["container==silent_bob"]' ...
+
+    # Run on a different node as all containers labelled with 'role=database':
+    docker run -e 'affinity:role!=database' ...
+
+    # Run on a different node to all containers with names starting with "foo":
+    docker run -e 'affinity:container!=foo*' ...
+
+    # Same, using a regular expression:
+    docker run -e 'affinity:container!=/^foo/' ...
+
+Note: At the time of writing the label syntax is in Docker Swarm
+[code](https://github.com/docker/swarm/blob/d9beef7/cluster/config.go#L83-L86),
+but not [documented](https://docs.docker.com/swarm/scheduler/filter/).
+
+
```

Comment:
Maybe we want to mention what to expect when a request cannot be fulfilled based on the affinity requirements (though the error is not the best at the moment). Something like this?

If no suitable server can be located based on the affinity requirements, the provisioning request will fail with the following error message:
    (DockerNoComputeResourcesError) No compute resources available.